setup.py

```diff
@@ -26,11 +26,42 @@
         '\nuse it. Set WALLY_ABI_PY_WHEEL_USE_LIB=static to link with the static library.\n',
         file=sys.stderr)
 
+def str_to_bool(str, if_none=ValueError, if_invalid=ValueError):
+    if str is None:
+        default = if_none
+    else:
+        lstr = str.lower()
+        if lstr in ('0', 'n', 'no', 'f', 'false', 'off'):
+            return False
+        elif lstr in ('1', 'y', 'yes', 't', 'true', 'on'):
+            return True
+        default = if_invalid
+    if isinstance(default, BaseException):
+        raise default('Invalid flag value: {}'.format(str))
+    return default
+
+def pkg_default(pkg, if_true, if_false=None):
+    b = str_to_bool(pkg, None, None)
+    return if_true if b == True else if_false if b == False else pkg
+
+PKG_SECP256K1 = pkg_default(
+    os.environ.get('WALLY_ABI_PY_WHEEL_USE_PKG_SECP256K1'), 'libsecp256k1_zkp')
+
```

Comment:
This doesn't need to be so general, less code means the actual build logic is clearer. Something like:

```
def _get_system_secp():
    secp_lib = os.environ.get('WALLY_ABI_PY_WHEEL_USE_PKG_SECP256K1')
    if not secp_lib or secp_lib.lower() in ('0', 'n', 'no', 'f', 'false', 'off'):
        return '' # Disabled
    if secp_lib.lower() in ('1', 'y', 'yes', 't', 'true', 'on'):
        return 'libsecp256k1_zkp' # Use the default system wide secp-zkp
    return secp_lib # Use the caller supplied secp library name


PKG_SECP256K1 = _get_system_secp() if USE_LIB != 'no' else ''
```

Should be sufficent. Then change `PKG_SECP256K1 is/is not None` to `if PKG_SECP256K1` /  `if not PKG_SECP256K1` below.

Reply:
That's not cleanly extensible to future dependencies that also might have pkg-config files.

Also, you have a bad habit of shortening "secp256k1" to "secp". There are other secp curves, you know.

Reply:
No problem, I understand. We are very close to the finish line now though, and I appreciate all the effort you have put in and your responsiveness to review.